lib/types/application/ApplicationFormatter.js

```diff
@@ -67,31 +108,61 @@ class ApplicationFormatter extends AbstractUi5Formatter {
 			}
 		}
 		const namespace = appId.replace(/\./g, "/");
-		log.verbose(`Namespace of project ${this._project.metadata.name} is ${namespace}`);
+		log.verbose(
+			`Namespace of project ${this._project.metadata.name} is ${namespace} (from manifest.appdescr_variant)`);
 		return namespace;
 	}
 
 	/**
-	 * Reads the projects manifest.json
+	 * Determine application namespace by checking manifest.json.
+	 * Any maven placeholders are resolved from the projects pom.xml
+	 *
+	 * @returns {string} Namespace of the project
+	 * @throws {Error} if namespace can not be determined
+	 */
+	async getNamespaceFromManifestAppDescVariant() {
+		const {content: manifest} = await this.getJson("manifest.appdescr_variant");
+		let appId;
+		// check for a proper sap.app/id in manifest.json to determine namespace
```

Comment:
// check for a proper id in manifest.appdescr_variant to determine namespace

Reply:
Done, thanks!